### ISSUE TITLE
IFRAME does not open Export dialog after a custom export button click.  

### ISSUE THREAD

**Issue by Andrey-Kachow** [avatarsdk/metaperson-loader-unity#8]
### Description
I encountered an issue with MetaPerson Creator IFrame, which does not display the export dialogue if the user clicks something outside the Iframe. For example, a custom export button, as in my case.

### Steps to Reproduce
1. Create an html file and paste the snippet below. It contains a minimal example
2. Replace XXX and YYY with the valid authentication details.
3. Open HTML in Chrome.
4. Follow the steps written on a self-explanatory button label.

**test.html**
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>MetaPerson Creator Web Sample</title>
     <style>
      * {
        margin: 0;
        padding: 0;
      }
    </style>
  </head>
  <body>
    <div style="display: flex, flex-direction: column; height: 96vh; width: 100vw;">
      <iframe id="editor_iframe" src="https://metaperson.avatarsdk.com/iframe.html" allow="fullscreen" frameborder="0" style="width: 100%; height: 90%;">
      </iframe>
      <div style="width: 100%; height: 10%;">
        <button id="custom-button" style="width: 100%; height: 100%; font-size: 3rem;" disabled>Authenticating...</button>
      </div>
    </div>
  </body>
  <script>
    let button;
    function onWindowMessage(evt) {
      if (evt.type === "message") {
        if (evt.data?.source === "metaperson_creator") {
          let data = evt.data;
          let evtName = data?.eventName;
          if (evtName == "metaperson_creator_loaded") {
            let authenticationMessage = {
              "eventName": "authenticate",
              "clientId": "XXX",
              "clientSecret": "YYY",
            };
            evt.source.postMessage(authenticationMessage, "*");
          }
          if (evtName == "authentication_status") {
            if (data.isAuthenticated) {
              button.textContent = "press \"show\" inside iframe..."
            } else {
              alert("failed to authenticate. Double check the credentials.");
            }
          }
          if (evtName == "action_availability_changed") {
            if (data.actionName == "avatar_export") {
              button.disabled = !data.isAvailable;
              if (data.isAvailable) {
                button.textContent = "now Click Me (I am the export button)"
              }
            }
          }
        }
      }
    }
    function buttonClick(event) {
      button.disabled = true;
      const iframe = document.getElementById("editor_iframe");
      const exportAvatarMessage = { "eventName": "export_avatar" };
      iframe.contentWindow.postMessage(exportAvatarMessage, "*");
      /*  iframe.focus(); // not in doc  */
      button.textContent = 'do not click anything else for 10 seconds...';
      setTimeout(() => {
        button.textContent = 'now click inside the editor'
      }, 10000);
    }
    document.addEventListener('DOMContentLoaded', function onDocumentReady() {
      window.addEventListener("message", onWindowMessage);
      button = document.getElementById('custom-button');
      button.addEventListener('click', buttonClick);
    });
  </script>
</html>
```

### Expected Behavior
I expected my export button to open the export dialogue window immediately.
Then, I hoped to receive the model_exported event that would launch my game.

### Actual Behavior
The IFRAME responds only once the user clicks back on it. Despite receiving the message from the client. That delayed the model exported event, and the game was not launched. Assuming the user can click the "Save and Play" button and then wait, they might get confused about why nothing happens.

### Environment
- Operating System: Windows
- Browser: Google Chrome

### Suggested Solution
The issue can be solved on the client side by simply adding `iframe.focus(),` so it is no longer a big issue for me.
However, I wonder if that is intended. I am informing you about that slightly dodgy behaviour, as the JS API does not mention that the iframe may lose focus. I hope that helps.

Thank you for your attention to this matter. Feel free to get in touch with me if you have any quesitons.

Best regards,
Andrey

**Comment by dledents** [comment]
Thank you for reporting this. It is a known issue.
In our [web sample](https://metaperson.avatarsdk.com/business.html) as a workaround we call `iframe.contentWindow.focus();` when user presses the **Export** or **Generate Avatar** button outside the iFrame. So, the same as you suggested.
We will look if we can do something on the MetaPerson Creator's side to avoid explicitly call `focus()` method.

**Comment by Andrey-Kachow** [comment]
Thank you for your response. I am glad that you are already aware of this and wish you good luck improving it. I appreciate your efforts and look forward to the resolution. I respect your priorities, so please let me know if you prefer to remain the issue open until resolution or to keep the project tidy and would like me to close the issue. Or feel free to close it yourself.
Best regards,
Andrey

**Comment by dledents** [comment]
The issue is fixed in the MetaPerson Creator v1.16.0: https://metaperson.avatarsdk.com/1.16.0/iframe.html
